public/pages/AnomalyCharts/containers/AnomalyDetailsChart.tsx

```diff
@@ -0,0 +1,388 @@
+}
+
+export const AnomalyDetailsChart = React.memo(
+  (props: AnomalyDetailsChartProps) => {
+    const dispatch = useDispatch();
+    const [anomalySummary, setAnomalySummary] = useState<AnomalySummary>(
+      INITIAL_ANOMALY_SUMMARY
+    );
+    const [showAlertsFlyout, setShowAlertsFlyout] = useState<boolean>(false);
+    const [alertAnnotations, setAlertAnnotations] = useState<any[]>([]);
+    const [isLoadingAlerts, setIsLoadingAlerts] = useState<boolean>(false);
+    const [totalAlerts, setTotalAlerts] = useState<number | undefined>(
+      undefined
+    );
+    const [alerts, setAlerts] = useState<MonitorAlert[]>([]);
+    const [zoomRange, setZoomRange] = useState<DateRange>({
+      ...props.dateRange,
+    });
+    const [zoomedAnomalies, setZoomedAnomalies] = useState<any[]>([]);
+
+    const DEFAULT_DATE_PICKER_RANGE = {
+      start: moment().subtract(7, 'days').valueOf(),
+      end: moment().valueOf(),
+    };
+
+    useEffect(() => {
+      const anomalies = prepareDataForChart(props.anomalies, zoomRange);
+      setZoomedAnomalies(anomalies);
+      setAnomalySummary(
+        !props.bucketizedAnomalies
+          ? getAnomalySummary(
+              filterWithDateRange(props.anomalies, zoomRange, 'plotTime')
+            )
+          : props.anomalySummary
+      );
+      setTotalAlerts(
+        filterWithDateRange(alerts, zoomRange, 'startTime').length
+      );
+    }, [props.anomalies, zoomRange, props.dateRange]);
+
+    useEffect(() => {
+      setZoomRange(props.dateRange);
+    }, [props.dateRange]);
+
+    const handleZoomRangeChange = (start: number, end: number) => {
+      setZoomRange({
+        startDate: start,
+        endDate: end,
+      });
+      props.onZoomRangeChange(start, end);
+    };
+
+    useEffect(() => {
+      async function getMonitorAlerts(
+        monitorId: string,
+        startDateTime: number,
+        endDateTime: number
+      ) {
+        try {
+          setIsLoadingAlerts(true);
+          const result = await dispatch(
+            searchAlerts(monitorId, startDateTime, endDateTime)
+          );
+          setIsLoadingAlerts(false);
+          setTotalAlerts(get(result, 'data.response.totalAlerts'));
+          const monitorAlerts = convertAlerts(result);
+          setAlerts(monitorAlerts);
+          const annotations = generateAlertAnnotations(monitorAlerts);
+          setAlertAnnotations(annotations);
+        } catch (err) {
+          console.error(`Failed to get alerts for monitor ${monitorId}`, err);
+          setIsLoadingAlerts(false);
+        }
+      }
+      if (
+        props.monitor &&
+        props.dateRange.startDate &&
+        // only load alert stats for non HC detector
+        props.isHCDetector != undefined &&
+        !props.isHCDetector
+      ) {
+        getMonitorAlerts(
+          props.monitor.id,
+          props.dateRange.startDate,
+          props.dateRange.endDate
+        );
+      }
+    }, [props.monitor, props.dateRange.startDate]);
```

Comment:
why only trigger this on `startDate` and not `endDate`?

Reply:
good catch. previous`getMonitorAlerts()` only accepts startDate, now it accepts start/end date now. Changing to include endDate